sdv/tabular/base.py

```diff
@@ -320,75 +319,69 @@ def _sample_batch(self, num_rows=None, max_retries=100, max_rows_multiplier=10,
             pandas.DataFrame:
                 Sampled data.
         """
+        if not batch_size_per_try:
+            batch_size_per_try = num_rows * 10
+
         sampled, num_valid = self._sample_rows(
             num_rows, conditions, transformed_conditions, float_rtol)
 
-        counter = 0
-        total_sampled = num_rows
+        counter = 1
         while num_valid < num_rows:
-            if counter >= max_retries:
+            if counter >= max_tries:
                 break
 
             remaining = num_rows - num_valid
-            valid_probability = (num_valid + 1) / (total_sampled + 1)
-            max_rows = num_rows * max_rows_multiplier
-            num_to_sample = min(int(remaining / valid_probability), max_rows)
-            total_sampled += num_to_sample
 
-            LOGGER.info('%s valid rows remaining. Resampling %s rows', remaining, num_to_sample)
+            LOGGER.info(f'{remaining} valid rows remaining. Resampling {batch_size_per_try} rows')
             sampled, num_valid = self._sample_rows(
-                num_to_sample, conditions, transformed_conditions, float_rtol, sampled
+                batch_size_per_try, conditions, transformed_conditions, float_rtol, sampled,
             )
 
             counter += 1
 
         return sampled.head(min(len(sampled), num_rows))
 
-    def _make_conditions_df(self, conditions, num_rows):
+    def _make_conditions_df(self, conditions):
```

Comment:
should the name change since this now returns a list?

Reply:
Sure, done!